Add files via upload

RidleyCJ opened the pull request:
Updated crystalBox.py to include tools for estimate/normalised Uiso based on element mass. A separate function is now included called loadCifNormUiso which replaces loadCif, correcting the Uiso values throughout.

mguthriem commented on crystalBox.py:
```diff
@@ -119,7 +121,39 @@ def loadCif(self):
 
         DeleteWorkspace(Workspace='tmp')
+    
+    def loadCifNormUiso(self):
```
I think it will be useful to retain the the original Uisos and corresponding Fsq/Int. More generally, I think it would be good to allow users to specify any arbitrary Uisos and update Fsq/Int. Now I think about it, it would be useful to allow users free control of x,y,z,f too. 

I would do this by retaining the original LoadCif, then create a new function `overrideCrystalStructure` (?) that, optionally, either calls your estimate_uiso or accepts user-specified values, then applies and recalculates the Fsq/Int

The original crystal structure should be kept and we probably need a `resetCrystalStructure` function to put things back as they were. Similar to the existing `resetLattice`

mguthriem commented on crystalBox.py:
```diff
@@ -236,13 +270,31 @@ def get_scatterers(self):
+        #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]
+        return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])      
+    
+    def estimate_uiso(self):
```
Should be called `estimateUiso`

mguthriem commented on crystalBox.py:
```diff
@@ -236,13 +270,31 @@ def get_scatterers(self):
         return scatterers
     
+    def set_scatterers(self):
+        scatterers = self.estimate_uiso()
```
This seems like it should be the place to generalise so that any crystal structure parmaeter can be modified. 

We should retain `estimate_uiso` as way to easily update only uisos using our standardised method.

Following comment above, setScatterers should automatically call process crystal to update FSq/Int to reflect the changes. 

Lastly, I try to systematically use "camelCase" so function name should be `setScatterers` (and I need to also update the copy/paset `get_Scatterers` that I stole from Zach)

mguthriem commented on crystalBox.py:
```diff
@@ -236,13 +270,31 @@ def get_scatterers(self):
+    
+    def estimate_uiso(self):
+        #sampleName = crys.Box('kdp')
+        CellContents=self.get_scatterers()
```
make `CellContents` a class attribute so user can access content. Also, rename to `cellContents`

mguthriem commented on crystalBox.py:
```diff
@@ -236,13 +270,31 @@ def get_scatterers(self):
+    def estimate_uiso(self):
+        #sampleName = crys.Box('kdp')
+        CellContents=self.get_scatterers()
+        OriginalCellContents = CellContents.copy()
```
originalCellContents should be exposed as a class attribute, but should be read-only i.e. it should be renamed `_originalCellContents` and made a python `property`

mguthriem reviewed (commented):
You will see from review I've been hanging around with coders for far too long.

Let me know how you want to proceed. If it makes more sense, we could use PR as a way for you to show me prototype code and then leave it to me to do the pedantic stuff, expand the scope, worry about other general users and the other nitty gritty I'm requesting here...

RidleyCJ commented:
You’re certainly guilty of project creep 😊

I’ll play around with it a bit, and send you some mods.

From: mguthriem ***@***.***>
Date: Friday, December 6, 2024 at 10:52 AM
To: mguthriem/crystalBox ***@***.***>
Cc: Ridley, Chris ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [mguthriem/crystalBox] Add files via upload (PR #7)

@mguthriem commented on this pull request.

You will see from review I've been hanging around with coders for far too long.

Let me know how you want to proceed. If it makes more sense, we could use PR as a way for you to show me prototype code and then leave it to me to do the pedantic stuff, expand the scope, worry about other general users and the other nitty gritty I'm requesting here...

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873481495&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=q9UasH7ExUTLDYyin_Wdsvh6xQ7YZBGrns9epRbJVXE&e=>:

> @@ -119,7 +121,39 @@ def loadCif(self):

         self.gamma_orig = self.gamma



         DeleteWorkspace(Workspace='tmp')

+

+    def loadCifNormUiso(self):

I think it will be useful to retain the the original Uisos and corresponding Fsq/Int. More generally, I think it would be good to allow users to specify any arbitrary Uisos and update Fsq/Int. Now I think about it, it would be useful to allow users free control of x,y,z,f too.

I would do this by retaining the original LoadCif, then create a new function overrideCrystalStructure (?) that, optionally, either calls your estimate_uiso or accepts user-specified values, then applies and recalculates the Fsq/Int

The original crystal structure should be kept and we probably need a resetCrystalStructure function to put things back as they were. Similar to the existing resetLattice

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873483379&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=FaoeMIRq-R6S0zwOEsxb2Y1OJXxdnPzBhicJnLxzjWU&e=>:

>          scatterers = [atm.split(' ') for atm in list(scatterers)]

         scatterers = [[val if val.isalpha() else float(val) \

                        for val in scatterer] for scatterer in scatterers]

         print(scatterers)

         return scatterers



+    def set_scatterers(self):

+        scatterers = self.estimate_uiso()

+        #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+        return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+    def estimate_uiso(self):

Should be called estimateUiso

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873488114&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=OZqgQENxoqhjERdBFFvPY3HZNZoaGeA4s1ULOKnJvYQ&e=>:

>          scatterers = [atm.split(' ') for atm in list(scatterers)]

         scatterers = [[val if val.isalpha() else float(val) \

                        for val in scatterer] for scatterer in scatterers]

         print(scatterers)

         return scatterers



+    def set_scatterers(self):

+        scatterers = self.estimate_uiso()

This seems like it should be the place to generalise so that any crystal structure parmaeter can be modified.

We should retain estimate_uiso as way to easily update only uisos using our standardised method.

Following comment above, setScatterers should automatically call process crystal to update FSq/Int to reflect the changes.

Lastly, I try to systematically use "camelCase" so function name should be setScatterers (and I need to also update the copy/paset get_Scatterers that I stole from Zach)

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873492298&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=JULi7gXamoKQrzgs162WLujlk7QGJllxPCTVWisOaYg&e=>:

>          scatterers = [atm.split(' ') for atm in list(scatterers)]

         scatterers = [[val if val.isalpha() else float(val) \

                        for val in scatterer] for scatterer in scatterers]

         print(scatterers)

         return scatterers



+    def set_scatterers(self):

+        scatterers = self.estimate_uiso()

+        #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+        return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+    def estimate_uiso(self):

+        #sampleName = crys.Box('kdp')

+        CellContents=self.get_scatterers()

make CellContents a class attribute so user can access content. Also, rename to cellContents

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873538310&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=MmbU9qP_I1kfG7Nb7qr0dnXI-jFyPJX78tUiEvnPp6s&e=>:

>          scatterers = [atm.split(' ') for atm in list(scatterers)]

         scatterers = [[val if val.isalpha() else float(val) \

                        for val in scatterer] for scatterer in scatterers]

         print(scatterers)

         return scatterers



+    def set_scatterers(self):

+        scatterers = self.estimate_uiso()

+        #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+        return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+    def estimate_uiso(self):

+        #sampleName = crys.Box('kdp')

+        CellContents=self.get_scatterers()

+        OriginalCellContents = CellContents.copy()

originalCellContents should be exposed as a class attribute, but should be read-only i.e. it should be renamed _originalCellContents and made a python property

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23pullrequestreview-2D2485076434&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=5q2-QgHRurIQ-hW_37As1npfd3UiwlnjzA6BxvIH_4E&e=>, or unsubscribe<https://urldefense.us/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_BJ5ONG4DSUBL3YCYRPP3AN32EHB2LAVCNFSM6AAAAABTDSLREGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIOBVGA3TMNBTGQ&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=Os4W3GgGIJXmmHY-uaoWlHvxAJBpnGETABB7aF3N0Jk&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

mguthriem commented:
Yes, sorry.

I think it makes sense to chunk my requests though.

I would happily pull a version that only implements the estimate_uiso functionality, but written in a way that I could subsequently add the functionality to change other things in the crystal structure.

M.

From: RidleyCJ ***@***.***>
Date: Friday, December 6, 2024 at 11:09 AM
To: mguthriem/crystalBox ***@***.***>
Cc: Guthrie, Malcolm ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [mguthriem/crystalBox] Add files via upload (PR #7)
You’re certainly guilty of project creep 😊

I’ll play around with it a bit, and send you some mods.

From: mguthriem ***@***.***>
Date: Friday, December 6, 2024 at 10:52 AM
To: mguthriem/crystalBox ***@***.***>
Cc: Ridley, Chris ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [mguthriem/crystalBox] Add files via upload (PR #7)

@mguthriem commented on this pull request.

You will see from review I've been hanging around with coders for far too long.

Let me know how you want to proceed. If it makes more sense, we could use PR as a way for you to show me prototype code and then leave it to me to do the pedantic stuff, expand the scope, worry about other general users and the other nitty gritty I'm requesting here...

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873481495&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=q9UasH7ExUTLDYyin_Wdsvh6xQ7YZBGrns9epRbJVXE&e=>:

> @@ -119,7 +121,39 @@ def loadCif(self):

self.gamma_orig = self.gamma



DeleteWorkspace(Workspace='tmp')

+

+ def loadCifNormUiso(self):

I think it will be useful to retain the the original Uisos and corresponding Fsq/Int. More generally, I think it would be good to allow users to specify any arbitrary Uisos and update Fsq/Int. Now I think about it, it would be useful to allow users free control of x,y,z,f too.

I would do this by retaining the original LoadCif, then create a new function overrideCrystalStructure (?) that, optionally, either calls your estimate_uiso or accepts user-specified values, then applies and recalculates the Fsq/Int

The original crystal structure should be kept and we probably need a resetCrystalStructure function to put things back as they were. Similar to the existing resetLattice

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873483379&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=FaoeMIRq-R6S0zwOEsxb2Y1OJXxdnPzBhicJnLxzjWU&e=>:

> scatterers = [atm.split(' ') for atm in list(scatterers)]

scatterers = [[val if val.isalpha() else float(val) \

for val in scatterer] for scatterer in scatterers]

print(scatterers)

return scatterers



+ def set_scatterers(self):

+ scatterers = self.estimate_uiso()

+ #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+ return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+ def estimate_uiso(self):

Should be called estimateUiso

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873488114&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=OZqgQENxoqhjERdBFFvPY3HZNZoaGeA4s1ULOKnJvYQ&e=>:

> scatterers = [atm.split(' ') for atm in list(scatterers)]

scatterers = [[val if val.isalpha() else float(val) \

for val in scatterer] for scatterer in scatterers]

print(scatterers)

return scatterers



+ def set_scatterers(self):

+ scatterers = self.estimate_uiso()

This seems like it should be the place to generalise so that any crystal structure parmaeter can be modified.

We should retain estimate_uiso as way to easily update only uisos using our standardised method.

Following comment above, setScatterers should automatically call process crystal to update FSq/Int to reflect the changes.

Lastly, I try to systematically use "camelCase" so function name should be setScatterers (and I need to also update the copy/paset get_Scatterers that I stole from Zach)

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873492298&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=JULi7gXamoKQrzgs162WLujlk7QGJllxPCTVWisOaYg&e=>:

> scatterers = [atm.split(' ') for atm in list(scatterers)]

scatterers = [[val if val.isalpha() else float(val) \

for val in scatterer] for scatterer in scatterers]

print(scatterers)

return scatterers



+ def set_scatterers(self):

+ scatterers = self.estimate_uiso()

+ #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+ return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+ def estimate_uiso(self):

+ #sampleName = crys.Box('kdp')

+ CellContents=self.get_scatterers()

make CellContents a class attribute so user can access content. Also, rename to cellContents

________________________________

In crystalBox.py<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23discussion-5Fr1873538310&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=MmbU9qP_I1kfG7Nb7qr0dnXI-jFyPJX78tUiEvnPp6s&e=>:

> scatterers = [atm.split(' ') for atm in list(scatterers)]

scatterers = [[val if val.isalpha() else float(val) \

for val in scatterer] for scatterer in scatterers]

print(scatterers)

return scatterers



+ def set_scatterers(self):

+ scatterers = self.estimate_uiso()

+ #return [f'{scatterer[0]} {scatterer[1]:.5f} {scatterer[2]:.5f} {scatterer[3]:.5f} {scatterer[4]:.0f} {scatterer[5]:.5f}' for scatterer in scatterers]

+ return "; ".join([f'{scatterer[0]} {scatterer[1]} {scatterer[2]} {scatterer[3]} {scatterer[4]} {scatterer[5]}' for scatterer in scatterers])

+

+ def estimate_uiso(self):

+ #sampleName = crys.Box('kdp')

+ CellContents=self.get_scatterers()

+ OriginalCellContents = CellContents.copy()

originalCellContents should be exposed as a class attribute, but should be read-only i.e. it should be renamed _originalCellContents and made a python property

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23pullrequestreview-2D2485076434&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=5q2-QgHRurIQ-hW_37As1npfd3UiwlnjzA6BxvIH_4E&e=>, or unsubscribe<https://urldefense.us/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_BJ5ONG4DSUBL3YCYRPP3AN32EHB2LAVCNFSM6AAAAABTDSLREGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIOBVGA3TMNBTGQ&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=6CezKWsettFZVk_ENtCKCKlsVO4UM6d-sWaDl7pijHc&m=kw8mo7-L0p7SgPrzMBHHxEG_VJJ9-67l-N60I1TukxPlDiCGhGayEnT3D_DsMLSH&s=Os4W3GgGIJXmmHY-uaoWlHvxAJBpnGETABB7aF3N0Jk&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v2/url?u=https-3A__github.com_mguthriem_crystalBox_pull_7-23issuecomment-2D2523604099&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=o3eBygIeKXRbWJt7Vra8lZbsjt0KL-0luHyCqFbyzuw&m=P5fSX-BTyUqfQ7vdBW2GgfqzXKl4Xct0Vm8A63S2c728ZzEBDdF2UOFE4N0f1T3q&s=LAZQHKzVUb6i3o0nb1Sax4bD7WqAuSzedaCy8wvlcWQ&e=>, or unsubscribe<https://urldefense.us/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AVUFLTATV7CKPRACYVTO42L2EHD3ZAVCNFSM6AAAAABTDSLREGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRTGYYDIMBZHE&d=DwMFaQ&c=v4IIwRuZAmwupIjowmMWUmLasxPEgYsgNI-O7C4ViYc&r=o3eBygIeKXRbWJt7Vra8lZbsjt0KL-0luHyCqFbyzuw&m=P5fSX-BTyUqfQ7vdBW2GgfqzXKl4Xct0Vm8A63S2c728ZzEBDdF2UOFE4N0f1T3q&s=yWOYo2Mx75QcipQgiiykmCr3dBywsdRawAojjrlxPWc&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

mguthriem reviewed (approved):
nice! nice!

I checked this works and was able to use `fixUiso` to change the Uiso. I tried Nickel, and diamond which get updated Uiso of 0.013 and 0.0288 respectively. We might want to tweak the equation used in `fixUiso` a little as I realise, it will give a value of 0.1 for hydrogen, which seems a bit too large?
